app/containers/System/containers/Firmware/index.js

```diff
@@ -85,6 +85,23 @@ const Firmware = () => {
     createdTimestamp,
     lastModifiedTimestamp
   ) => {
+    trackAssignmentData.getAllFirmwareTrackAssignment.forEach((element, index) => {
+      if (element.modelId === modelId) {
+        const prevFirmwareVersionRecordId =
+          trackAssignmentData.getAllFirmwareTrackAssignment[index].firmwareVersionRecordId;
+        const prevTrackRecordId =
+          trackAssignmentData.getAllFirmwareTrackAssignment[index].trackRecordId;
+        if (prevFirmwareVersionRecordId !== firmwareVersionRecordId) {
+          deleteTrackAssignment({
+            variables: {
+              firmwareTrackId: prevTrackRecordId,
+              firmwareVersionId: prevFirmwareVersionRecordId,
+            },
+          });
+        }
+      }
+    });
```

Comment:
instead of all this unnecessary work, just pass the previous firmwareVersionRecordId to handleUpdateTrackAssignment